File: kadai.py
def isValid(s: str) -> bool:
    if len(s) % 2 != 0:
        return False

    bracket_map = {")": "(", "}": "{", "]": "["}
    bracket_stack = []
    for c in s:
        if c not in bracket_map:
            bracket_stack.append(c)
        else:
            if not bracket_stack or bracket_stack.pop() != bracket_map.get(c):
                return False
    return not bracket_stack


s = "()"
print(isValid(s))  # True
s = "({)}"
print(isValid(s))  # False

File: kadai.js
const isValid = (s) => {
  if (s.length % 2 != 0) return false;
  var bracketStack = [];
  for (const c of s) {
    switch (c) {
      case "(":
        bracketStack.push(")");
        break;
      case "{":
        bracketStack.push("}");
        break;
      case "[":
        bracketStack.push("]");
        break;
      default:
        if (!bracketStack.length || bracketStack.pop() != c) return false;
        break;
    }
  }
  return !bracketStack.length;
};
let s = "()";
console.log(isValid(s)); // true
s = "({)}";
console.log(isValid(s)); // false
